File: leetcode/js/row-with-maximum-ones.js
/**
 * @param {number[][]} mat
 * @return {number[]}
 */
var rowAndMaximumOnes = function (mat) {
  let idxOfRow = 0;
  let maxCount = 0;

  mat.forEach((row, idx) => {
    const count = row.filter((num) => num === 1).length;

    if (maxCount < count) {
      idxOfRow = idx;
      maxCount = count;
    }
  });

  return [idxOfRow, maxCount];
};

File: leetcode/rust/row-with-maximum-ones.rs
impl Solution {
    pub fn row_and_maximum_ones(mat: Vec<Vec<i32>>) -> Vec<i32> {
        let mut idx_of_row = 0;
        let mut max_count = 0;

        for (idx, row) in mat.iter().enumerate() {
            let count = row.iter().sum();

            if max_count < count {
                idx_of_row = idx;
                max_count = count;
            }
        }

        vec![idx_of_row as i32, max_count]
    }
}
